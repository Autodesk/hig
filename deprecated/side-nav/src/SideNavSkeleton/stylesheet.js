import { constants } from "../constants";

export default function stylesheet(props) {
  const { stylesheet: customStylesheet } = props;
  const styles = {
    boxSizing: "border-box",
    paddingLeft: "24px",
    paddingTop: "48px",
    width: constants.sideNavMaxWidth,
  };

  if (customStylesheet) {
    return customStylesheet(styles, props);
  }

  return styles;
}
