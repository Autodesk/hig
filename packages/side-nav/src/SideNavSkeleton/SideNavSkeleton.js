import React, { Component } from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import SkeletonItem from "@hig/skeleton-item";
import stylesheet from "./stylesheet";

const skeletonItemStyles = Object.freeze([
  {
    maxWidth: "180px",
    marginBottom: "48px"
  },
  {
    maxWidth: "180px",
    marginBottom: "24px"
  },
  {
    maxWidth: "180px",
    marginBottom: "24px"
  },
  {
    maxWidth: "124px",
    marginBottom: "48px"
  },
  {
    maxWidth: "180px",
    marginBottom: "24px"
  },
  {
    maxWidth: "180px",
    marginBottom: "24px"
  },
  {
    maxWidth: "180px",
    marginBottom: "24px"
  },
  {
    maxWidth: "180px",
    marginBottom: "24px"
  },
  {
    maxWidth: "124px"
  }
]);

export default class SideNavSkeleton extends Component {
  static propTypes = {
    /** Function to modify the component's styles */
    stylesheet: PropTypes.func
  };

  render() {
    const { stylesheet: customStylesheet, ...otherProps } = this.props;
    const { className } = otherProps;

    return (
      <div
        className={cx([
          css(stylesheet({ stylesheet: customStylesheet, ...this.props })),
          className
        ])}
      >
        {skeletonItemStyles.map((style, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <SkeletonItem key={index} {...style} />
        ))}
      </div>
    );
  }
}
