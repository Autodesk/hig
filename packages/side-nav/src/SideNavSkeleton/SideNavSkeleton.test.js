import React from "react";
import renderer from "react-test-renderer";

import SideNavSkeleton from "./SideNavSkeleton";

describe("side-nav/SideNavSkeleton", () => {
  describe("snapshot tests", () => {
    const cases = [
      {
        description: "renders with no props",
        props: {},
      },
    ];

    cases.forEach(({ description, props: { children, ...otherProps } }) => {
      it(description, () => {
        const wrapper = (
          <SideNavSkeleton {...otherProps}>{children}</SideNavSkeleton>
        );
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
