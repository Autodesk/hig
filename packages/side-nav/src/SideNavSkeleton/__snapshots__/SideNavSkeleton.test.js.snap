// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/SideNavSkeleton snapshot tests renders with no props 1`] = `
<div
  className="hig__side-nav-skeleton hig--light-theme"
>
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "48px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "24px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "24px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "48px",
        "maxWidth": "124px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "24px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "24px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "24px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": "24px",
        "maxWidth": "180px",
      }
    }
  />
  <div
    className="hig__skeleton-item hig--light-theme"
    style={
      Object {
        "height": undefined,
        "marginBottom": undefined,
        "maxWidth": "124px",
      }
    }
  />
</div>
`;
